Delivered-To: openglfreak@gmail.com
Received: by 2002:a54:364d:0:0:0:0:0 with SMTP id s13csp2348990ect;
        Sat, 27 Mar 2021 12:48:11 -0700 (PDT)
X-Google-Smtp-Source: ABdhPJy0w3E5kKBf6I/jZLPv2wZqlb+Dlnufwlhq5dUn2hdJqjXeQ+TJLnrUyzMOsWutj2wM0wOh
X-Received: by 2002:a92:c702:: with SMTP id a2mr15704248ilp.219.1616874491381;
        Sat, 27 Mar 2021 12:48:11 -0700 (PDT)
ARC-Seal: i=1; a=rsa-sha256; t=1616874491; cv=none;
        d=google.com; s=arc-20160816;
        b=Yn8nI8HuL+nvdNUmzd/gy7pXU/JgHMBd1mEK1yIs/iZSBlw90vWCfQbQzp9SPuLqg6
         zalagSozDlsmNKltq6nPixDqhlOzXOwv6XpJa1K9xmjBLdRzOC+bWDLC0aLV7/BVGApt
         lxvTVRQvWO6oWQnvofAQSdGp38wPL0lVYA9YOz1a9kE/DBbSfZKJJ5HlGhMzzq5B/jmM
         2mBhweya6jbc7FHsrf/dRGxNlzoM8Tf/9jC8c9QfWXqggcliIDOCdxR6pnlzZfcDAaR6
         pc4NS/lRUKYv8BDyrUHfOQKHuxVlVDqEmAA7HvAq8cH/beCwUSYTHwiH+ZJkaj3fawZp
         LFQA==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=google.com; s=arc-20160816;
        h=sender:errors-to:cc:list-subscribe:list-help:list-post:list-archive
         :list-unsubscribe:list-id:precedence:content-transfer-encoding
         :mime-version:references:in-reply-to:message-id:date:subject:to:from
         :dkim-signature:dkim-signature;
        bh=VsivhKdMa5kc477NLb9MzvT7pQi2oknqzN6qgL2/1hw=;
        b=zyUKPmTBNlQnStPtyhCTIYpPZdtlFYZDUb9ZGYozdPJlnS2PGXNldSOrQFRRdPSzUS
         vUhO9Z2NU45+2jZNqG7JDDzPteUG05ZHTtLIHLgp7G2buI73xgg7Jafn/ZV5x708t7mW
         rhWY9zTKfZNFhxxaNXv+zMz7hQSHZLJq4n9SeL5r19zy/YwmckuxpPPQJQOEspax5zi/
         Blg774651+Q6HX1e3R58uz6245ibvtWV7+Ve+UXPoneM52XNhVFW6sA81g7oKRfPMrWW
         hT1qu7Qv4yG22dL+v/wxbIlNEX9wba3GKW9NjADw5CVrUa52RzH6f5MhdtNjT4gefx2+
         ZF6A==
ARC-Authentication-Results: i=1; mx.google.com;
       dkim=pass (test mode) header.i=@winehq.org header.s=20170406 header.b=eQ55KmXJ;
       dkim=fail header.i=@gmail.com header.s=20161025 header.b=LCFpTBph;
       spf=pass (google.com: domain of wine-devel-bounces@winehq.org designates 4.4.81.124 as permitted sender) smtp.mailfrom=wine-devel-bounces@winehq.org
Return-Path: <wine-devel-bounces@winehq.org>
Received: from winehq.org (winehq.org. [4.4.81.124])
        by mx.google.com with ESMTPS id b5si11011406jat.99.2021.03.27.12.48.11
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Sat, 27 Mar 2021 12:48:11 -0700 (PDT)
Received-SPF: pass (google.com: domain of wine-devel-bounces@winehq.org designates 4.4.81.124 as permitted sender) client-ip=4.4.81.124;
Authentication-Results: mx.google.com;
       dkim=pass (test mode) header.i=@winehq.org header.s=20170406 header.b=eQ55KmXJ;
       dkim=fail header.i=@gmail.com header.s=20161025 header.b=LCFpTBph;
       spf=pass (google.com: domain of wine-devel-bounces@winehq.org designates 4.4.81.124 as permitted sender) smtp.mailfrom=wine-devel-bounces@winehq.org
DKIM-Signature: v=1; a=rsa-sha256; q=dns/txt; c=relaxed/relaxed; d=winehq.org;
	 s=20170406; h=Sender:Cc:List-Subscribe:List-Help:List-Post:List-Archive:
	List-Unsubscribe:List-Id:Content-Transfer-Encoding:MIME-Version:References:
	In-Reply-To:Message-Id:Date:Subject:To:From:Reply-To:Content-Type:Content-ID:
	Content-Description:Resent-Date:Resent-From:Resent-Sender:Resent-To:Resent-Cc
	:Resent-Message-ID:List-Owner;
	bh=VsivhKdMa5kc477NLb9MzvT7pQi2oknqzN6qgL2/1hw=; b=eQ55KmXJvew250+FbtH2UN2Plh
	w4/72i/mNKerBomwtJu+wFveI+ns0ldVbyqGNtlgZe82rmyGnUSf7q1uZuQCyEb80odFm1xX1UU7x
	Go5ovrtsO3GpS9rAfVWVV+s/w/BMAS5WzWovKFinkCJEgVk2nG6PgwOANO4cJMtJpq9c=;
Received: from localhost ([::1] helo=wine.codeweavers.com)
	by winehq.org with esmtp (Exim 4.92)
	(envelope-from <wine-devel-bounces@winehq.org>)
	id 1lQEun-0006AE-F3; Sat, 27 Mar 2021 14:48:09 -0500
Received: from mail-wm1-f45.google.com ([209.85.128.45])
 by winehq.org with esmtps (TLS1.3:ECDHE_RSA_AES_128_GCM_SHA256:128)
 (Exim 4.92) (envelope-from <mark.harmstone@gmail.com>)
 id 1lQEub-00060T-7W
 for wine-devel@winehq.org; Sat, 27 Mar 2021 14:47:58 -0500
Received: by mail-wm1-f45.google.com with SMTP id
 z6-20020a1c4c060000b029010f13694ba2so4656235wmf.5
 for <wine-devel@winehq.org>; Sat, 27 Mar 2021 12:47:57 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=gmail.com; s=20161025;
 h=sender:from:to:cc:subject:date:message-id:in-reply-to:references
 :mime-version:content-transfer-encoding;
 bh=VsivhKdMa5kc477NLb9MzvT7pQi2oknqzN6qgL2/1hw=;
 b=LCFpTBphDmbm5Ktv3O9SilRIgxhe87vXnKINm62DKVFaPabJN92aN3bFY8S6E0FljS
 9VwYafvZylV0TkOaaX5ZgDRpiFPUhokY99APGmOXbLu2RuLNDPHYPX+MwbwXqBiHj1Ir
 3MEd1j0P1bBp+R22NuEXM/tEte1fmXYicuc1dW52Z9VOsBWxk7IEbzq7VTqVEQqJjYDI
 nn/qyPcP4e756mnDTxQU5QmZfkW0FLqdhGvc4GBY4kJRwhYUlPt5sGL0/s3lZBmsluic
 TAsot1f7ZTdecJ2bobAanawHRLdj5oPUckWCWbFALfau2eXNltRsqo/8iuT9KZw+itYu
 EdNQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=1e100.net; s=20161025;
 h=x-gm-message-state:sender:from:to:cc:subject:date:message-id
 :in-reply-to:references:mime-version:content-transfer-encoding;
 bh=VsivhKdMa5kc477NLb9MzvT7pQi2oknqzN6qgL2/1hw=;
 b=ItZxLbgRn5CALfapSk+soyEkXdMNa7Id6m/P1GhL/RDoLmztcOX76d5/SZMuIoBf0j
 OGZtfQ+OjxldAxKRuojNK3HwMWSNeu6YnH1rJbukRHI4fVONzHgDJIQFeqEdVf7IB0CM
 BjeiGs787F3yyqS7ly2O7HkltmDFkBzYkGNCoas+M0wchEVMSxNfsSQJ+vPoMXGMJGEL
 M2HKL6TpG/VyX5rXL+EmQRpz0vnKc9jIXnaJFuo6gFFpA57jKGFycDMkBoiTk38MpVFI
 rAl+zgJdYgiJaGCc+HanTIJ5FMHfymtnt+T597490gU/Bt+7GwrnEzYzpcko+VUq4O3d
 /LOQ==
X-Gm-Message-State: AOAM533kFPUyU92/JE7Jm2EBOUv38mf7KX6x0S38xh1We4h2qeFrSMIU
 xFuSXU+yiap1VLNzoQ/tq+CX9Dr3yNE=
X-Received: by 2002:a05:600c:acf:: with SMTP id
 c15mr18571126wmr.124.1616874475985; 
 Sat, 27 Mar 2021 12:47:55 -0700 (PDT)
Received: from localhost.localdomain ([2a02:8010:64ea:0:fad1:11ff:fead:57db])
 by smtp.gmail.com with ESMTPSA id
 64sm16187141wmz.7.2021.03.27.12.47.55
 (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
 Sat, 27 Mar 2021 12:47:55 -0700 (PDT)
From: Mark Harmstone <mark@harmstone.com>
To: wine-devel@winehq.org
Subject: [PATCH 04/10] ntdll: Add stubs for LdrAddLoadAsDataTable and
 LdrRemoveLoadAsDataTable.
Date: Sat, 27 Mar 2021 19:47:31 +0000
Message-Id: <20210327194737.13026-4-mark@harmstone.com>
X-Mailer: git-send-email 2.26.3
In-Reply-To: <20210327194737.13026-1-mark@harmstone.com>
References: <20210327194737.13026-1-mark@harmstone.com>
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
X-Spam-Score: 1.8
X-Spam-Report: Action: no action Symbol: RCVD_VIA_SMTP_AUTH(0.00)
 Symbol: TO_DN_SOME(0.00) Symbol: R_SPF_ALLOW(-0.20)
 Symbol: R_MISSING_CHARSET(2.50)
 Symbol: BROKEN_CONTENT_TYPE(1.50) Symbol: RCVD_COUNT_THREE(0.00)
 Symbol: DKIM_TRACE(0.00) Symbol: RCPT_COUNT_TWO(0.00)
 Symbol: FORGED_SENDER(0.30) Symbol: MIME_TRACE(0.00)
 Symbol: FREEMAIL_ENVFROM(0.00) Symbol: ASN(0.00)
 Symbol: TAGGED_FROM(0.00) Symbol: FROM_NEQ_ENVFROM(0.00)
 Symbol: DWL_DNSWL_NONE(0.00) Symbol: ARC_NA(0.00)
 Symbol: R_DKIM_ALLOW(-0.20) Symbol: BAYES_HAM(-3.00)
 Symbol: FROM_HAS_DN(0.00) Symbol: MIME_GOOD(-0.10)
 Symbol: PREVIOUSLY_DELIVERED(0.00) Symbol: DMARC_NA(0.00)
 Symbol: SENDER_REP_HAM(0.00) Symbol: TO_MATCH_ENVRCPT_SOME(0.00)
 Symbol: MID_CONTAINS_FROM(1.00) Symbol: RCVD_IN_DNSWL_NONE(0.00)
 Symbol: NEURAL_HAM(-0.00) Symbol: RWL_MAILSPIKE_POSSIBLE(0.00)
 Symbol: RCVD_TLS_ALL(0.00)
 Message-ID: 20210327194737.13026-4-mark@harmstone.com
X-BeenThere: wine-devel@winehq.org
X-Mailman-Version: 2.1.29
Precedence: list
List-Id: Wine Developer's List <wine-devel.winehq.org>
List-Unsubscribe: <https://www.winehq.org/mailman/options/wine-devel>,
 <mailto:wine-devel-request@winehq.org?subject=unsubscribe>
List-Archive: <http://www.winehq.org/pipermail/wine-devel/>
List-Post: <mailto:wine-devel@winehq.org>
List-Help: <mailto:wine-devel-request@winehq.org?subject=help>
List-Subscribe: <https://www.winehq.org/mailman/listinfo/wine-devel>,
 <mailto:wine-devel-request@winehq.org?subject=subscribe>
Cc: Mark Harmstone <mark@harmstone.com>
Errors-To: wine-devel-bounces@winehq.org
Sender: "wine-devel" <wine-devel-bounces@winehq.org>
X-Spam-Score: -8.2
X-Spam-Report: Action: no action
 Symbol: RCVD_VIA_SMTP_AUTH(0.00)
 Symbol: TO_DN_SOME(0.00)
 Symbol: R_MISSING_CHARSET(2.50)
 Symbol: REPLY(-4.00)
 Symbol: BROKEN_CONTENT_TYPE(1.50)
 Symbol: RCVD_COUNT_THREE(0.00)
 Symbol: RCPT_COUNT_TWO(0.00)
 Symbol: MAILLIST(-0.20)
 Symbol: RCVD_NO_TLS_LAST(0.10)
 Symbol: FORGED_RECIPIENTS_MAILLIST(0.00)
 Symbol: MIME_TRACE(0.00)
 Symbol: FROM_NEQ_ENVFROM(0.00)
 Symbol: BAYES_HAM(-3.00)
 Symbol: ARC_NA(0.00)
 Symbol: FROM_HAS_DN(0.00)
 Symbol: FREEMAIL_ENVRCPT(0.00)
 Symbol: TAGGED_RCPT(0.00)
 Symbol: MIME_GOOD(-0.10)
 Symbol: PREVIOUSLY_DELIVERED(0.00)
 Symbol: HAS_LIST_UNSUB(-0.01)
 Symbol: MID_CONTAINS_FROM(1.00)
 Symbol: NEURAL_HAM(-0.00)
 Symbol: WHITELIST_SENDER_DOMAIN(-6.00)
 Symbol: FORGED_SENDER_MAILLIST(0.00)
 Message-ID: 20210327194737.13026-4-mark@harmstone.com

Signed-off-by: Mark Harmstone <mark@harmstone.com>
---
 dlls/ntdll/mui.c      | 28 ++++++++++++++++++++++++++++
 dlls/ntdll/ntdll.spec |  2 ++
 include/winternl.h    |  2 ++
 3 files changed, 32 insertions(+)

diff --git a/dlls/ntdll/mui.c b/dlls/ntdll/mui.c
index 11111111111..11111111111 100644
--- a/dlls/ntdll/mui.c
+++ b/dlls/ntdll/mui.c
@@ -20,6 +20,8 @@
 
 #include <stdarg.h>
 
+#include "ntstatus.h"
+#define WIN32_NO_STATUS
 #include "winternl.h"
 #include "wine/debug.h"
 
@@ -510,3 +512,29 @@ BOOLEAN WINAPI RtlLCIDToCultureName( LCID lcid, PUNICODE_STRING string )
 
     return FALSE;
 }
+
+/***********************************************************************
+ *           LdrAddLoadAsDataTable (NTDLL.@)
+ *
+ * Called when a data-only PE image is mapped.
+ */
+NTSTATUS NTAPI LdrAddLoadAsDataTable( PVOID module, PWSTR file_path, SIZE_T size, HANDLE handle )
+{
+    FIXME("(%p, %s, %lx, %p)\n", module, debugstr_w(file_path), size, handle);
+
+    return STATUS_NOT_IMPLEMENTED;
+}
+
+/***********************************************************************
+ *           LdrRemoveLoadAsDataTable (NTDLL.@)
+ *
+ * Called when a PE image is unloaded, that had been mapped for data
+ * access only.
+ */
+NTSTATUS NTAPI LdrRemoveLoadAsDataTable( PVOID init_module, PVOID *base_module, PSIZE_T size,
+                                         ULONG flags )
+{
+    FIXME("(%p, %p, %p, %x)\n", init_module, base_module, size, flags);
+
+    return STATUS_NOT_IMPLEMENTED;
+}
diff --git a/dlls/ntdll/ntdll.spec b/dlls/ntdll/ntdll.spec
index 11111111111..11111111111 100644
--- a/dlls/ntdll/ntdll.spec
+++ b/dlls/ntdll/ntdll.spec
@@ -74,6 +74,7 @@
 # @ stub LdrAccessOutOfProcessResource
 @ stdcall LdrAccessResource(long ptr ptr ptr)
 @ stdcall LdrAddDllDirectory(ptr ptr)
+@ stdcall LdrAddLoadAsDataTable(ptr ptr long ptr)
 @ stdcall LdrAddRefDll(long ptr)
 # @ stub LdrAlternateResourcesEnabled
 # @ stub LdrCreateOutOfProcessImage
@@ -105,6 +106,7 @@
 @ stdcall LdrQueryProcessModuleInformation(ptr long ptr)
 @ stdcall LdrRegisterDllNotification(long ptr ptr ptr)
 @ stdcall LdrRemoveDllDirectory(ptr)
+@ stdcall LdrRemoveLoadAsDataTable(ptr ptr ptr long)
 @ stdcall LdrResolveDelayLoadedAPI(ptr ptr ptr ptr ptr long)
 @ stub LdrSetAppCompatDllRedirectionCallback
 @ stdcall LdrSetDefaultDllDirectories(long)
diff --git a/include/winternl.h b/include/winternl.h
index 11111111111..11111111111 100644
--- a/include/winternl.h
+++ b/include/winternl.h
@@ -3889,6 +3889,7 @@ NTSYSAPI NTSTATUS  WINAPI DbgUiWaitStateChange(DBGUI_WAIT_STATE_CHANGE*,LARGE_IN
 NTSYSAPI void      WINAPI DbgUserBreakPoint(void);
 NTSYSAPI NTSTATUS  WINAPI LdrAccessResource(HMODULE,const IMAGE_RESOURCE_DATA_ENTRY*,void**,PULONG);
 NTSYSAPI NTSTATUS  WINAPI LdrAddDllDirectory(const UNICODE_STRING*,void**);
+NTSYSAPI NTSTATUS  WINAPI LdrAddLoadAsDataTable(PVOID, PWSTR, SIZE_T, HANDLE);
 NTSYSAPI NTSTATUS  WINAPI LdrAddRefDll(ULONG,HMODULE);
 NTSYSAPI NTSTATUS  WINAPI LdrDisableThreadCalloutsForDll(HMODULE);
 NTSYSAPI NTSTATUS  WINAPI LdrFindEntryForAddress(const void*, PLDR_DATA_TABLE_ENTRY*);
@@ -3907,6 +3908,7 @@ NTSYSAPI NTSTATUS  WINAPI LdrQueryImageFileExecutionOptions(const UNICODE_STRING
 NTSYSAPI NTSTATUS  WINAPI LdrQueryProcessModuleInformation(RTL_PROCESS_MODULES*, ULONG, ULONG*);
 NTSYSAPI NTSTATUS  WINAPI LdrRegisterDllNotification(ULONG,PLDR_DLL_NOTIFICATION_FUNCTION,void*,void**);
 NTSYSAPI NTSTATUS  WINAPI LdrRemoveDllDirectory(void*);
+NTSYSAPI NTSTATUS  WINAPI LdrRemoveLoadAsDataTable(PVOID, PVOID*, PSIZE_T, ULONG);
 NTSYSAPI NTSTATUS  WINAPI LdrSetDefaultDllDirectories(ULONG);
 NTSYSAPI NTSTATUS  WINAPI LdrSetDllDirectory(const UNICODE_STRING*);
 NTSYSAPI void      WINAPI LdrShutdownProcess(void);
